pykwasm/src/pykwasm/kdist/wasm-semantics/numeric.md

```diff
@@ -136,18 +136,18 @@ Note: The actual `ctz` operator considers the integer 0 to have *all* zero-bits,
     rule ITYPE . ctz    I1 => < ITYPE > #if I1 ==Int 0 #then #width(ITYPE) #else #ctz(I1) #fi
     rule ITYPE . popcnt I1 => < ITYPE > #popcnt(I1)
 
-    syntax Int ::= #minWidth ( Int ) [function]
-                 | #ctz      ( Int ) [function]
-                 | #popcnt   ( Int ) [function]
+    syntax Int ::= #minWidth ( Int ) [function, total]
+                 | #ctz      ( Int ) [function, total]
+                 | #popcnt   ( Int ) [function, total]
  // -------------------------------------------
-    rule #minWidth(0) => 0
-    rule #minWidth(N) => 1 +Int #minWidth(N >>Int 1)                                 requires N =/=Int 0
+    rule #minWidth(N) => 0                                                           requires N <=Int 0
+    rule #minWidth(N) => 1 +Int #minWidth(N >>Int 1)                                 requires N >Int 0
 
-    rule #ctz(0) => 0
-    rule #ctz(N) => #if N modInt 2 ==Int 1 #then 0 #else 1 +Int #ctz(N >>Int 1) #fi  requires N =/=Int 0
+    rule #ctz(N) => 0                                                                requires N <=Int 0
+    rule #ctz(N) => #if N modInt 2 ==Int 1 #then 0 #else 1 +Int #ctz(N >>Int 1) #fi  requires N >Int 0
```

Comment:
AFAICT this function was already total before the change (the right-shift will eventually produce a `1` in the LSb unless the number is indeed `0`).
Should we keep the `modInt` or use `&Int`?
```suggestion
    rule #ctz(N) => 0                      requires N ==Int 0
    rule #ctz(N) => 0                      requires N &Int 1 ==Int 1
    rule #ctz(N) => 1 +Int #ctz(N >>Int 1) [owise]
```